cpu/cc2538/radio/cc2538-rf-netdev.c

```diff
@@ -0,0 +1,375 @@
+static int get_state(netdev_t *dev, netdev_state_t *state)
+{
+    if (dev != &cc2538_netdev) {
+        return -ENODEV;
+    }
+
+    return netdev_state;
+}
+
+static int set_state(netdev_t *dev, netdev_state_t state)
+{
+    if (dev != &cc2538_netdev) {
+        return -ENODEV;
+    }
+
+    switch (state) {
+        case NETDEV_STATE_POWER_OFF:
+        case NETDEV_STATE_POWER_SLEEP:
+        case NETDEV_STATE_POWER_IDLE:
+            cc2538_off();
+            break;
+
+        case NETDEV_STATE_RX_MODE:
+            cc2538_switch_to_rx();
+            break;
+
+        case NETDEV_STATE_PROMISCUOUS_MODE:
+            cc2538_set_monitor(false);
+            cc2538_switch_to_rx();
+            break;
+
+        default:
+        case NETDEV_STATE_TX_BURST:
+            return -ENOTSUP;
+    }
+
+    netdev_state = state;
+    return 0;
+}
+
+static void event(netdev_t *dev, uint32_t event_type)
+{
+    (void)dev;
+    (void)event_type;
+}
+
+static netdev_802154_tx_status_t load_tx(
+        netdev_t *dev,
+        netdev_802154_pkt_kind_t kind,
+        netdev_802154_node_addr_t *dest,
+        int use_long_addr,
+        int wants_ack,
+        netdev_hlist_t *upper_layer_hdrs,
+        void *buf,
+        unsigned int len)
+{
+    mac_header_t hdr;
+    uint_fast8_t hdr_size;
+    static uint_fast8_t sequence_nr = 0;
+
+    if (dev != &cc2538_netdev) {
+        return NETDEV_802154_TX_STATUS_NO_DEV;
+    }
+
+    /* FCS : frame version 0, we don't manage security,
+       nor batches of packets */
+    switch (kind) {
+        case PACKET_KIND_BEACON:
+        case PACKET_KIND_DATA:
+        case PACKET_KIND_ACK:
+            hdr.type = kind;
+            break;
+
+        default:
+            return NETDEV_802154_TX_STATUS_INVALID_PARAM;
+    }
+
+    if (wants_ack) {
+        hdr.type |= WANTS_ACK;
+    }
+
+    if (use_long_addr) {
+        hdr.addr_type = ADDR_TYPE_LONG;
+        hdr.long_addrs.dest_addr = HTONLL(dest->long_addr);
+        hdr.long_addrs.src_addr  = HTONLL(cc2538_get_address_long());
+        hdr_size = 3 + sizeof(long_addrs_t);
+    }
+    else {
+        hdr.addr_type = ADDR_TYPE_SHORT;
+        uint_fast16_t src_pan = cc2538_get_pan();
+
+        if (dest->pan.id == src_pan) {
+            hdr.type |= INTRA_PAN_COMPRESSION;
+            hdr.short_addrs_compr.dest_pan  = HTONS(dest->pan.id);
+            hdr.short_addrs_compr.dest_addr = HTONS(dest->pan.addr);
+            hdr.short_addrs_compr.src_addr  = HTONS(cc2538_get_address());
+            hdr_size = 3 + sizeof(short_addrs_compr_t);
+        }
+        else {
+            hdr.short_addrs.dest_pan  = HTONS(dest->pan.id);
+            hdr.short_addrs.dest_addr = HTONS(dest->pan.addr);
+            hdr.short_addrs.src_pan   = HTONS(src_pan);
+            hdr.short_addrs.src_addr  = HTONS(cc2538_get_address());
+            hdr_size = 3 + sizeof(short_addrs_t);
+        }
+    }
+
+    /* total frame size */
+    if (1 + hdr_size + len > CC2538_RF_FIFO_SIZE) {
```

Comment:
Please document `1` or provide a `#define`. I'd guess it is for `CC2538_AUTOCRC_LEN`?